### PR TITLE
Bump linux FIPS builds to use go1.16.2b7 release

### PR DESCRIPTION
**Pull request by a-palchikov**
Update Go runtime to 1.16.2 and bump the boringcrypto version correspondingly for linux FIPS builds

### CODE REVIEW

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -97,7 +97,7 @@ ENV GOGOPROTO_ROOT ${GOPATH}/src/github.com/gogo/protobuf
 RUN (curl -L -o /tmp/${PROTOC_TARBALL} https://github.com/google/protobuf/releases/download/v${PROTOC_VER}/${PROTOC_TARBALL} && \
      cd /tmp && unzip /tmp/${PROTOC_TARBALL} -d /usr/local && \
      rm /tmp/${PROTOC_TARBALL})
-RUN (go get -u github.com/gogo/protobuf/proto github.com/gogo/protobuf/protoc-gen-gogo github.com/gogo/protobuf/gogoproto golang.org/x/tools/cmd/goimports && \
+RUN (git clone https://github.com/gogo/protobuf.git ${GOPATH}/src/github.com/gogo/protobuf && go install golang.org/x/tools/cmd/goimports@latest && \
```
Comment (webvictim) [inline build.assets/Dockerfile:100]: What's the reason for the change here?

File: build.assets/Makefile [hunk 2]
```diff
@@ -11,13 +11,17 @@ NOROOT=-u $$(id -u):$$(id -g)
...
+
 OS ?= linux
 ARCH ?= amd64
-RUNTIME ?= go1.15.5
+RUNTIME ?= go1.16.2
 
 UID := $$(id -u)
 GID := $$(id -g)
 
+BORINGCRYPTO_RUNTIME=$(RUNTIME)b7
```
Comment (webvictim) [inline build.assets/Makefile:23]: Move this up below `RUNTIME`

```suggestion
RUNTIME ?= go1.16.2
BORINGCRYPTO_RUNTIME=$(RUNTIME)b7

UID := $$(id -u)
GID := $$(id -g)
```

File: build.assets/Makefile [hunk 3]
```diff
@@ -11,13 +11,17 @@ NOROOT=-u $$(id -u):$$(id -g)
 KUBECONFIG ?=
 TEST_KUBE ?=
 
+SHELL=bash
```
Comment (webvictim) [inline build.assets/Makefile:14]: The `busybox` that `docker:git` and similar images for Drone use only has busybox `sh` out of the box - you can install `bash` with `apk --no-cache add bash` but why do we need it?

Comment (a-palchikov) [reply build.assets/Makefile]: I already removed it for lack of a better thought. The reason I added it initially is that I could not build specific `release` targets as these are bash-flavored so it was the knee jerk action.

File: build.assets/Dockerfile (hunk 1 above)
Comment (a-palchikov) [reply build.assets/Dockerfile:100]: This has to do with go1.16 changes to `go get` and I pulled these from my other [PR](https://github.com/gravitational/teleport/pull/5935/files#diff-1e5924d92807f9242cbbcdd715eb80a2237f28d8f4642a1d987df5b9c04a0811R100)

**Comment by webvictim** [comment]
We should test a tag build before we merge this.

This is what I usually do:
- Update `VERSION` in the `Makefile` to the next `dev` tag (`7.0.0-dev.5` here)
- Run `make version.go` and commit the changes
- `git tag v7.0.0-dev.5 && git push origin v7.0.0-dev.5`

**Review by webvictim — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Comment by webvictim** [comment]
Tag build tested. All good, except BoringCrypto 1.16.2 won't compile from source on CentOS 6:

```
Step 10/15 : RUN mkdir -p /go-bootstrap && cd /go-bootstrap && curl https://dl.google.com/go/${GO_BOOTSTRAP_RUNTIME}.linux-amd64.tar.gz \| tar xz &&     mkdir -p /opt && cd /opt && curl https://go-boringcrypto.storage.googleapis.com/${BORINGCRYPTO_RUNTIME}.src.tar.gz \| tar xz &&     cd /opt/go/src && GOROOT_BOOTSTRAP=/go-bootstrap/go ./make.bash &&     rm -rf /go-bootstrap &&     mkdir -p /go/src/github.com/gravitational/teleport &&     chmod a+w /go &&     chmod a+w /var/lib &&     chmod a-w / &&     /opt/go/bin/go version
---> Running in 8d19ce45ea46
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current | 124s
Dload  Upload   Total   Spent    Left  Speed
0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  54  112M   54 61.9M    0     0  65.1M      0  0:00:01 --:--:--  0:00:01 70.7M  96  112M   96  108M    0     0  55.5M      0  0:00:02  0:00:01  0:00:01 57.7M 100  112M  100  112M    0     0  53.8M      0  0:00:02  0:00:02 --:--:-- 55.8M
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Dload  Upload   Total   Spent    Left  Speed
0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  99 22.6M   99 22.5M    0     0  26.9M      0 --:--:-- --:--:-- --:--:-- 28.7M 100 22.6M  100 22.6M    0     0  26.8M      0 --:--:-- --:--:-- --:--:-- 28.6M
Building Go cmd/dist using /go-bootstrap/go. (go1.9.7 linux/amd64)
Building Go toolchain1 using /go-bootstrap/go.
Building Go bootstrap cmd/go (go_bootstrap) using Go toolchain1.
Building Go toolchain2 using go_bootstrap and Go toolchain1.
# crypto/internal/boring
src/crypto/internal/boring/goboringcrypto_linux_amd64.syso: In function `bcm_redirector_getauxval':
bcm.c:(.text+0xb40ed): undefined reference to `getauxval'
collect2: ld returned 1 exit status
go tool dist: FAILED: /opt/go/pkg/tool/linux_amd64/go_bootstrap install -gcflags=all= -ldflags=all= -i cmd/asm cmd/cgo cmd/compile cmd/link: exit status 2
The command '/bin/sh -c mkdir -p /go-bootstrap && cd /go-bootstrap && curl https://dl.google.com/go/${GO_BOOTSTRAP_RUNTIME}.linux-amd64.tar.gz \| tar xz &&     mkdir -p /opt && cd /opt && curl https://go-boringcrypto.storage.googleapis.com/${BORINGCRYPTO_RUNTIME}.src.tar.gz \| tar xz &&     cd /opt/go/src && GOROOT_BOOTSTRAP=/go-bootstrap/go ./make.bash &&     rm -rf /go-bootstrap &&     mkdir -p /go/src/github.com/gravitational/teleport &&     chmod a+w /go &&     chmod a+w /var/lib &&     chmod a-w / &&     /opt/go/bin/go version' returned a non-zero code: 2
make: *** [Makefile:141: buildbox-centos6-fips] Error 2
```

For now I'm going to merge this so we stop getting build failures on `master`. We can try to fix this next week.
